tests/Credentials/AssumeRoleWithWebIdentityCredentialProviderTest.php

```diff
@@ -396,4 +399,168 @@ public function testCanDisableInvalidIdentityTokenRetries()
             unlink($tokenPath);
         }
     }
+
+    /**
+     * Tests region precedence: config > env var > fallback
+     * @dataProvider regionPrecedenceProvider
+     */
+    public function testRegionPrecedence(
+        ?string $configRegion,
+        ?string $envRegion,
+        string $expectedRegion,
+        bool $expectNotice
+    ): void
+    {
+        $originalRegion = getenv('AWS_REGION');
+
+        if ($envRegion !== null) {
+            putenv("AWS_REGION={$envRegion}");
+        } else {
+            putenv("AWS_REGION");
+        }
+
+        $tokenFile = tempnam(sys_get_temp_dir(), 'token');
+        file_put_contents($tokenFile, 'test-token-content');
+
+        $config = [
+            'RoleArn' => self::SAMPLE_ROLE_ARN,
+            'WebIdentityTokenFile' => $tokenFile,
+        ];
+
+        if ($configRegion !== null) {
+            $config['region'] = $configRegion;
+        }
+
+        if ($expectNotice) {
+            $this->expectNotice();
+            $this->expectNoticeMessage(
+                'NOTICE: STS client created without explicit `region` configuration.'
+            );
+        }
+
+        $provider = new AssumeRoleWithWebIdentityCredentialProvider($config);
+
+        // Check region
+        $reflection = new \ReflectionClass($provider);
+        $clientProperty = $reflection->getProperty('client');
+        $clientProperty->setAccessible(true);
+        $stsClient = $clientProperty->getValue($provider);
+
+        $this->assertEquals($expectedRegion, $stsClient->getRegion());
+
+        // Cleanup
+        unlink($tokenFile);
+
+        // Restore original AWS_REGION value
+        if ($originalRegion !== false) {
+            putenv("AWS_REGION={$originalRegion}");
+        } else {
+            putenv("AWS_REGION");
+        }
```

Comment:
I think this way of restoring `AWS_REGION` is not safe. For example, if a failure occurred then the variable will not be restored. I recommend to use a try {} finally {} block for it.